openmp/libomptarget/src/OffloadRTL.cpp

```diff
@@ -20,25 +20,37 @@
 extern void llvm::omp::target::ompt::connectLibrary();
 #endif
 
-__attribute__((constructor(101))) void init() {
+static std::mutex PluginMtx;
+static std::atomic<uint32_t> RefCount = 0;
```

Comment:
Why is this atomic, it seems all accesses are guarded by PluginMtx?

Reply:
Thought that myself, was erring on the side of caution in case someone somewhere wants to read it, but likely a non-issue.

Reply:
Rather put the counter into PM, and wrap PM into ProtectedObj? That makes it clear. I dislike complexity that is not needed, but I get why you went here. Still, this way one thinks the RefCount is accessed concurrently. ProtectedObj has benefits over a standalone Mtx too.

Reply:
That seems more complicated to me, since we'd need to somehow put the reference count into it, whereas as far as I'm aware the protected object simply wraps a mutex around a single object.